library/src/fy_library/mixins/property/template_model/using_parsed_fy_py_file_fy__for_setter_fy.py

```diff
@@ -7,24 +7,36 @@
 
 property template_model: BaseTemplateModel using parsed_fy_py_file__for_setter:
     property parsed_fy_py_file
+    property parsed_fy_py_files_map_by_key
 """
 
 import abc
 from functools import cached_property
 
 from fy_library.domain.fy_py_template_models import BaseTemplateModel
+from fy_library.flows.create_template_model_using_parsed_fy_py_file.main_fy import (
+    CreateTemplateModelUsingParsedFyPyFile_Flow,
+)
 from fy_library.mixins.property.parsed_fy_py_file.abc_fy import (
     ParsedFyPyFile_PropertyMixin_ABC,
 )
 
+from fy_library.mixins.property.parsed_fy_py_files_map_by_key.abc_fy import (
+    ParsedFyPyFilesMapByKey_PropertyMixin_ABC,
+)
+
 
 # fy:start ===>>>
 class TemplateModel_UsingParsedFyPyFile_ForSetter_PropertyMixin(
     # Property_mixins
     ParsedFyPyFile_PropertyMixin_ABC,
+    ParsedFyPyFilesMapByKey_PropertyMixin_ABC,
     abc.ABC,
 ):
     @cached_property
     def _template_model(self) -> BaseTemplateModel:
         # fy:end <<<===
-        return self._parsed_fy_py_file.template_model
+        return CreateTemplateModelUsingParsedFyPyFile_Flow(
```

Comment:
Let's create the template model here. Then you will not need to propagate `parsed_fy_py_files_map_by_key` via setters since you do not need it.

That means removing `parsed_fy_py_files_map_by_key` from all places in this PR it was added to.

Reply:
This is what I talk about - https://github.com/execution-flows/fy/pull/387/files#diff-74783a0491e5aaacea419f32c7d60d28405555ce56b8d300309cae879a1c4626R98-R101